### ISSUE TITLE
Text_tokenizer messes encoding

### ISSUE THREAD

**Issue by LasWin** [rstudio/keras#947]
Hi! I've noticed that text_tokenizer-function does not work with non-ASCII letters. I noticed it when trying to use my own pre-trained Finnish word embeddings as part of a text classifying model. 

See example:

> library(keras)
> 
> example <- "This text has some Weird SKÄNDINÄVIÄN letters ööö jåå äh"
> 
> tokenizer <- text_tokenizer(num_words = 1000) %>% fit_text_tokenizer(example)
> 
> names(tokenizer$word_index)

> [1] "this"            "text"            "has"             "some"            "weird"           "skÃ¤ndinÃ¤viÃ¤n"
>  [7] "letters"         "Ã¶Ã¶Ã¶"          "jÃ¥Ã¥"           "Ã¤h"     
> 

> names(tokenizer$word_index) <- gsub("Ã¤","ä",names(tokenizer$word_index))

> Error in py_set_attr_impl(x, name, value) : 
>   Evaluation error: UnicodeDecodeError: 'utf-8' codec can't decode byte 0xf6 in position 0: invalid start byte.

This results with all the ä's and ö's being replaced with a mess. And the gsub gives a Python UnicodeDecodeError. 

I can bypass this issue by replacing all the ä's and ö's both in the word embeddings and in the text being passed to Keras tokenizer but this adds an unnecessary layer to the code. I've tried all manners of encodings before and after feeding the text to text_tokenizer but to no avail.

I noticed the same problem was reported earlier in issue [349](https://github.com/rstudio/keras/issues/349) regarding Chinese letters, however that seems to have been closed without a fix. 

The issue here is only with Keras, my other text analysis packages (eg udpipe, text2vec) work brilliantly with UTF-8 encoding.

**Comment by skeydan** [comment]
Hello, this works fine for me in RStudio:

```

> library(keras)
> example <- "This text has some Weird SKÄNDINÄVIÄN letters ööö jåå äh"
> tokenizer <- text_tokenizer(num_words = 1000) %>% fit_text_tokenizer(example)
> names(tokenizer$word_index)
 [1] "this"         "text"         "has"          "some"        
 [5] "weird"        "skändinäviän" "letters"      "ööö"         
 [9] "jåå"          "äh"    

```

What is your system encoding? Can you try changing the file encoding in RStudio

Tools > Global Options > Code > Saving > "Default text encoding" 

to UTF8 and see if that helps?

**Comment by LasWin** [comment]
I tried that to no avail. Also tried to save the R-files with UTF-8-encoding, nevermind all the encoding tricks I did to text itself. Nothing seems to work :/   

My locale is Finnish_Finland.1252.

Since this is all about classification I can live with replacing umlauts. I guess not much you can do when you cannot reproduce the error.

Encodings are truly the biggest enemy of non-English coders, I don't even want to imagine how many hours I've spent cursing at them :D

**Comment by skeydan** [comment]
Yeah, unfortunately I don't seem to be able to reproduce - I can't set that locale on Linux, and when instead I set LANG to `fi_FI.UTF-8` it works fine...

I suppose you tried setting the locale to `en_US.UTF-8` before starting RStudio, right?

In any case, there is a new and better alternative to Keras text tokenizer, which is the text vectorization layer, as explained here: https://tensorflow.rstudio.com/tutorials/beginners/basic-ml/tutorial_basic_text_classification/

It's one of several very nice recent additions (actually as of today you'll need to install the nightly version of TensorFlow). See also: 

https://blogs.rstudio.com/tensorflow/posts/2019-11-27-gettingstarted-2020/
